Mathlib/RingTheory/LocalRing/MaximalIdeal/Basic.lean

```diff
@@ -99,3 +106,33 @@ end LocalRing
 
 theorem LocalRing.maximalIdeal_eq_bot {R : Type*} [Field R] : LocalRing.maximalIdeal R = ⊥ :=
   LocalRing.isField_iff_maximalIdeal_eq.mp (Field.toIsField R)
+
+section Nilrad_max_localization
+
+open Ideal
+
+variable {R : Type*} [CommSemiring R] {S : Type*} [CommSemiring S] [Algebra R S] {M : Submonoid R}
+
+instance (priority := 100) LocalRing.of_nilradical_isMaximal [h : (nilradical R).IsMaximal] :
```

Comment:
```suggestion
-- TODO: Make this an `instance`
theorem LocalRing.of_nilradical_isMaximal [h : (nilradical R).IsMaximal] :
```
Since the `instance` isn't working, just leave it a theorem and maybe someone else will figure out how to convert it to an instance if/when it's needed.